include/tty.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2018 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_TTY_H_
+#define ZEPHYR_INCLUDE_TTY_H_
+
+#include <sys/types.h>
+#include <zephyr/types.h>
+#include <kernel.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct tty_serial {
+	struct device *uart_dev;
+
+	struct k_sem rx_sem;
+	u8_t *rx_ringbuf;
+	u32_t rx_ringbuf_sz;
+	u16_t rx_get, rx_put;
+	s32_t rx_timeout;
+
+	struct k_sem tx_sem;
+	u8_t *tx_ringbuf;
+	u32_t tx_ringbuf_sz;
+	u16_t tx_get, tx_put;
+	s32_t tx_timeout;
+};
+
+/**
+ * @brief Initialize buffered serial port (classically known as tty).
+ *
+ * "tty" device provides buffered, interrupt-driver access to an
+ * underlying UART device.
+ *
+ * @param tty tty device structure to initialize
+ * @param uart_dev underlying UART device to use (should support
```

Comment:
As I understand, tty API should cover also other transport ways than UART. This should be common API to support UART, CDC ACM, RTT etc. and allow to handle all serial-like devices in the same manner. Like ttyS, ttyACM, ttyUSB in linux

Reply:
> As I understand, tty API should cover also other transport ways than UART.

Whether that's the case or not is still subject to discussion (ongoing at https://github.com/zephyrproject-rtos/zephyr/issues/10512).

The purpose of this patch is very simple (trivial, indeed, as the label suggests) - to move some arguably misplaced declarations into a header more suitable for them. Per the definition of Trivial-labeled patch, there're no changes to semantics or behavior.

Reply:
Ok then